fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java

```diff
@@ -233,9 +233,8 @@ public boolean equals(Object o) {
             return false;
         }
         GroupExpression that = (GroupExpression) o;
-        // TODO: add relation id to UnboundRelation
-        if (plan instanceof UnboundRelation) {
-            return false;
+        if (this.plan instanceof UnboundRelation) {
+            return plan.equals(that.plan);
```

Comment:
remove this. fix `computeLogicalProperties` in `AbstractLogicalPlan`. Let it return `UnboundLogicalProperties.INSTANCE` when current object is instance of `Unbound`